plugins/resources/unixfilesystem/src/unixfilesystem.cpp

```diff
@@ -280,7 +367,29 @@ irods::error unix_file_mkdir_r(const std::string& path, mode_t mode)
 /// @brief interface to notify of a file registration
 irods::error unix_file_registered(irods::plugin_context& _ctx)
 {
-    // NOOP
+    if (is_operating_in_detached_mode(_ctx.prop_map())) {
+        irods::file_object_ptr fco = boost::dynamic_pointer_cast<irods::file_object>(_ctx.fco());
+        std::string file_path = fco->physical_path();
+
+        std::string vault_path;
+        _ctx.prop_map().get<std::string>(irods::RESOURCE_PATH, vault_path);
+
+        if (!file_path.starts_with(vault_path)) {
```

Comment:
This can lead to incorrect results. Prefer `has_prefix()` when checking paths for prefixes.
https://github.com/irods/irods/blob/02b03c67ff428959a758a17fdbcbd25001b4e39d/lib/core/include/irods/rodsPath.h#L77-L90

Reply:
I did that because that is what Codacy said to do.  I'll just ignore Codacy from now on.

Reply:
I get what you are saying but I figured the paths would have been normalized before this point.  Change has been made.